### PR TITLE
PXC-4466: PXC 8.4.2 refresh - Q3 2024

### PR DESCRIPTION
**Pull request by kamil-holubicki**
https://perconadev.atlassian.net/browse/PXC-4466

1. Fixed problems reported by check_src tool
2. Allow run MTR tests for the server built with WITH_WSREP=OFF

### CODE REVIEW

File: check_src/whitelist/sql_admin.cc [hunk 1]
```diff
@@ -1,5 +1,3 @@
 <   /* table can be nullptr if we failed with wsrep_toi_replication() */
 <   if (table && table->table) table->table->invalidate_dict();
->   if (table->table) table->table->invalidate_dict();
-<   WSREP_NBO_1ST_PHASE_END;
-< 
+>   if (table->table) table->table->invalidate_dict();
```
Comment (venkatesh-prasad-v) [inline check_src/whitelist/sql_admin.cc:3]: Does this mean that `WSREP_NBO_1ST_PHASE_END` was removed from PXC sources?

File: check_src/whitelist/sql_partition_admin.cc [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-<     if (thd->mdl_context.upgrade_shared_lock(ticket, MDL_EXCLUSIVE, timeout)) {
-<       WSREP_NBO_1ST_PHASE_END;
->     if (thd->mdl_context.upgrade_shared_lock(ticket, MDL_EXCLUSIVE, timeout))
-<     }
-<           first_table->db, first_table->table_name, &table_def)) {
-<     WSREP_NBO_1ST_PHASE_END;
->           first_table->db, first_table->table_name, &table_def))
-<   }
-< 
-<   WSREP_NBO_1ST_PHASE_END;
```
Comment (venkatesh-prasad-v) [inline check_src/whitelist/sql_partition_admin.cc:14]: Does this mean that WSREP_NBO_1ST_PHASE_END was removed from PXC sources?

File: check_src/whitelist/sql_admin.cc (hunk 1 above)
Comment (kamil-holubicki) [reply check_src/whitelist/sql_admin.cc:3]: No, it is already wrapped with `ifdef WITH_WSREP`

File: check_src/whitelist/sql_partition_admin.cc (hunk 2 above)
Comment (kamil-holubicki) [reply check_src/whitelist/sql_partition_admin.cc:14]: No, it is already wrapped with `ifdef WITH_WSREP`

**Review by venkatesh-prasad-v — APPROVED** [review]
LGTM

**Pull request merged**